Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -257,16 +249,10 @@
 #    endif // SENTRY_HAS_UIKIT
 )
 {
-    if (isProfiling && [hub.getClient.options isContinuousProfilingEnabled]) {
-        if (hub.getClient.options.profiling.lifecycle != SentryProfileLifecycleTrace) {
-            SENTRY_TEST_FATAL(@"Expected trace profile v2 lifecycle configuration.");
-            return;
-        }
-
-        if (![SentryContinuousProfiler isCurrentlyProfiling]) {
-            SENTRY_TEST_FATAL(
-                @"Expected a continuous profiler to be running for this configuration.");
-        }
+    if (isProfiling && [hub.getClient.options isContinuousProfilingV2Enabled] &&
+        [hub.getClient.options isProfilingCorrelatedToTraces]) {
+        SENTRY_LOG_DEBUG(@"Stopping tracking root span tracer with profilerReferenceId %@",
+            transaction.trace.profilerReferenceID.sentryIdString);
```

Comment:
There was no good reason to even do anything for invalid combinations, it would come down to user error I think, which we already warn on in logs. Simply make the check for the right configuration and don't mess with anything otherwise here.